doc/md/fundamentals/2-types/12-advanced-types.md

```diff
@@ -25,25 +25,24 @@ Even though `p1` and `p2` are distinct objects, they are considered equal becaus
 
 Generic types are used to define type parameters that work with multiple data types, commonly used in [functions](https://internetcomputer.org/docs/motoko/fundamentals/types/functions), [classes](https://internetcomputer.org/docs/motoko/fundamentals/types/objects-classes), and data structures.
 
-```motoko no-repl
+```motoko
 // Generic function
 func identity<T>(x : T) : T {
   return x;
-}
+};
 
-let num = identity<Nat>(42);  // num is Nat
-let txt = identity<Text>("Hello");  // txt is Text
+identity<Nat>(42);  // num is Nat
 ```
 
 A generic class can store any type while maintaining type safety:
 
-```motoko no-repl
+```motoko
 class Box<T>(value : T) {
   public func get() : T { value };
-}
+};
 
 let intBox = Box<Nat>(10);
-let textBox = Box<Text>("Hello");
+intBox.get();
 ```
 
 ## Recursive types
```

Comment:
This example below (List<T>) actually illustrates two things. recursive types and parameterized types.

Maybe it would be good to first introduce recursive types
List = ?(Nat, List)
And implement  a recursive function reverse.

and then, to introduce parametirized types, generalize to

List<T> = ?(T, List<T>)

and implement generic reverse (using a recursive function).